### PR TITLE
[JAX] Schedule EP dispatch/combine on XLA collective stream

### PR DESCRIPTION
**Pull request by phu0ngng**
# Description

Pin the TE-JAX Expert Parallelism dispatch/combine ops to XLA's high-priority collective stream so the scheduler serializes them with native collectives instead of overlapping.

## Type of change

- [x] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/jax/cpp_extensions/ep.py [hunk 1]
```diff
@@ -19,12 +19,17 @@
 import jax
 import jax.numpy as jnp
 from jax import dtypes, ffi
+from jax.experimental.compute_on import compute_on
```
Comment (jberchtold-nvidia) [inline transformer_engine/jax/cpp_extensions/ep.py]: What JAX version was this added in? Should we guard with something like
```
try:
   from jax.experimental.compute_on import compute_on
except ImportError:
   def compute_on(stream_name: str):
       """ compute_on shim for older JAX versions """
       # Make @compute_on annotation do nothing to the annotated function
       return lambda f: f
```

Comment (phu0ngng) [reply transformer_engine/jax/cpp_extensions/ep.py]: Good suggestion for versioning.

The `import compute_on` is available since 2024, I think, so this check won't help here. 
I added a guard for JAX 0.10.0.

**Review by jberchtold-nvidia — APPROVED** [review]
LGTM pending CI, thanks!
